File: ru/codes/ruby/chapter_stack_and_queue/stack.rb
=begin
File: stack.rb
Created Time: 2024-04-06
Author: Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
=end

### Driver Code ###
if __FILE__ == $0
  # Инициализировать стек
  # В Ruby нет встроенного класса стека, поэтому Array можно использовать как стек
  stack = []

  # Поместить элемент в стек
  stack << 1
  stack << 3
  stack << 2
  stack << 5
  stack << 4
  puts "стек stack = #{stack}"

  # Получить верхний элемент стека
  peek = stack.last
  puts "вершина стекаэлемент peek = #{peek}"

  # Извлечь элемент из стека
  pop = stack.pop
  puts "Элемент, извлеченный из стека, pop = #{pop}"
  puts "stack после извлечения = #{stack}"

  # Получить длину стека
  size = stack.length
  puts "Длина стека size = #{size}"

  # Проверить, пуста ли структура
  is_empty = stack.empty?
  puts "стекпуст ли = #{is_empty}"
end

File: ru/codes/javascript/chapter_stack_and_queue/stack.js
/**
 * File: stack.js
 * Created Time: 2022-12-04
 * Author: S-N-O-R-L-A-X (snorlax.xu@outlook.com)
 */

/* Driver Code */
/* Инициализация стека */
// В JavaScript нет встроенного класса стека, поэтому Array можно использовать как стек
const stack = [];

/* Помещение элемента в стек */
stack.push(1);
stack.push(3);
stack.push(2);
stack.push(5);
stack.push(4);
console.log('Стек stack =', stack);

/* Доступ к верхнему элементу стека */
const peek = stack[stack.length - 1];
console.log('Верхний элемент peek =', peek);

/* Извлечение элемента из стека */
const pop = stack.pop();
console.log('Извлеченный элемент pop =', pop);
console.log('stack после извлечения =', stack);

/* Получение длины стека */
const size = stack.length;
console.log('Длина стека size =', size);

/* Проверка на пустоту */
const isEmpty = stack.length === 0;
console.log('Пуст ли стек =', isEmpty);
